chat-plugins/trivia.js

```diff
@@ -61,26 +75,69 @@ var Trivia = (function () {
 		this.cap = cap;
 		this.prize = (cap - 5) / 15 + 2;
 		this.phase = 'signup';
-		this.participants = {};
+		this.participants = new Map();
 		this.curQs = [];
 		this.curA = [];
 		this.sleep = null;
 
-		switch (mode) {
-		case 'timer':
-			this.askedAt = 0;
-			/* fall through */
-		case 'number':
-			this.responders = {};
-			break;
+		if (mode !== 'first') {
+			if (mode === 'timer') this.askedAt = 0;
+			this.correctResponders = 0;
 		}
 	}
 
-	// trivia question/answer loop methods
+	Trivia.prototype.addParticipant = function (output, user) {
+		if (this.phase !== 'signup') return output.sendReply('There is no trivia game in its signup phase.');
+		if (this.participants.has(user)) return output.sendReply('You have already signed up for this trivia game.');
+
+		var latestIp = user.latestIp;
+		for (var participant, participantsIterator = this.participants.keys(); !!(participant = participantsIterator.next().value);) { // replace with for-of loop once available
```

Comment:
Better written as
´´´var iterator = this.participants.keys();
while (!iterator.next().done) {
var participant = iterator.value;
// do stuff
}